CMakeLists.txt

```
@@ -232,13 +239,14 @@ endif()
 # Getting this wrong results in building large parts of the stablehlo
 # project that we don't actually depend on. Further some of those parts
 # do not even compile on all platforms.
-if (TORCH_MLIR_ENABLE_STABLEHLO)
-  set(STABLEHLO_BUILD_EMBEDDED ON)
-  set(STABLEHLO_ENABLE_BINDINGS_PYTHON ON)
-  add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/externals/stablehlo
-    ${CMAKE_CURRENT_BINARY_DIR}/stablehlo
-    EXCLUDE_FROM_ALL)
-  include_directories(${CMAKE_CURRENT_SOURCE_DIR}/externals/stablehlo)
+# Only configure StableHLO if it isn't provided from a top-level project
+if (TORCH_MLIR_ENABLE_STABLEHLO AND NOT TORCH_MLIR_USE_EXTERNAL_STABLEHLO)
+    set(STABLEHLO_BUILD_EMBEDDED ON)
+    set(STABLEHLO_ENABLE_BINDINGS_PYTHON ON)
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/externals/stablehlo
+      ${CMAKE_CURRENT_BINARY_DIR}/stablehlo
+      EXCLUDE_FROM_ALL)
+    include_directories(${CMAKE_CURRENT_SOURCE_DIR}/externals/stablehlo)
```

```suggestion
  set(STABLEHLO_BUILD_EMBEDDED ON)
  set(STABLEHLO_ENABLE_BINDINGS_PYTHON ON)
  add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/externals/stablehlo
    ${CMAKE_CURRENT_BINARY_DIR}/stablehlo
    EXCLUDE_FROM_ALL)
  include_directories(${CMAKE_CURRENT_SOURCE_DIR}/externals/stablehlo)
```
To keep it consistent the indent should be as before thus effectively only changing the if condition.

---

Done. Thanks for pointing it out.